core/src/main/java/org/elasticsearch/action/support/replication/TransportReplicationAction.java

```diff
@@ -708,7 +711,8 @@ private void failReplicaIfNeeded(String index, int shardId, Throwable t) {
          */
         public ReplicationPhase(ShardIterator originalShardIt, ReplicaRequest replicaRequest, Response finalResponse,
                                 ClusterStateObserver observer, ShardRouting originalPrimaryShard,
-                                InternalRequest internalRequest, ActionListener<Response> listener, Releasable indexShardReference) {
+                                InternalRequest internalRequest, ActionListener<Response> listener, Releasable indexShardReference,
+                                ShardStateActionListener shardStateListener) {
```

Comment:
For now I exposed it as a parameter to hook into for the testing that I added in [`TransportReplicationActionTests`](https://github.com/jasontedor/elasticsearch/commit/b46e42bc1f91ca1a17705f78bb0435612cf41d53#diff-3988756cc764d148f763c9d0cb740c61R360). However this might not be necessary and we can go with the approach that you suggest depending on how we respond to your [comment](https://github.com/elastic/elasticsearch/pull/14295#discussion_r43093730) regarding moving the initial testing out of `TransportReplicationActionTests`. I'll return to this after addressing the other item.